packages/client/src/components/app/forms/InnerForm.svelte

```diff
@@ -408,16 +408,38 @@
     }
   }
 
-  const handleScrollToField = ({ field }) => {
-    const fieldId = get(getField(field)).fieldState.fieldId
-    const label = document.querySelector(`label[for="${fieldId}"]`)
-    document.getElementById(fieldId).focus({ preventScroll: true })
-    label.scrollIntoView({ behavior: "smooth" })
+  const handleScrollToField = ({ field, block }) => {
+    if (!field.fieldState) {
+      field = get(getField(field))
+    }
+    const fieldId = field.fieldState.fieldId
+    const fieldElement = document.getElementById(fieldId)
+    fieldElement.focus({ preventScroll: true })
+    if (block === "start") {
+      const label = document.querySelector(`label[for="${fieldId}"]`)
+      label.scrollIntoView({ behavior: "smooth" })
+    } else {
+      fieldElement.scrollIntoView({
+        behavior: "smooth",
+        block,
+      })
+    }
   }
 
   // Action context to pass to children
   const actions = [
-    { type: ActionTypes.ValidateForm, callback: formApi.validate },
+    {
+      type: ActionTypes.ValidateForm,
+      callback: () => ({
+        fields: fields
+          .filter(field => get(field).step === get(currentStep))
+          .map(field => ({
+            field: get(field),
+            valid: get(field).fieldApi.validate(),
+          })),
+        valid: formApi.validate(),
+      }),
+    },
```

Comment:
I think we'd be better just moving the "scroll to the first invalid field" behaviour into the formApi.validate function itself. It would make sense for that to internally handle it. That also means we can remove this custom logic here, and we can also remove the custom return value logic in the button actions file where we manually invoke scrolling.

Let's just automatically scroll to the first invalid field before we return the validation result, inside the InnerForm validate function itself.